Assets/MixedRealityToolkit/InputSystem/Sources/GestureInputSource.cs

```diff
@@ -0,0 +1,191 @@
+
+        /// <summary>
+        /// Hold action to use when events are raised.
+        /// </summary>
+        public static MixedRealityInputAction HoldAction { get; set; } = MixedRealityInputAction.None;
+
+        /// <summary>
+        /// Manipulation action to use when events are raised.
+        /// </summary>
+        public static MixedRealityInputAction ManipulationAction { get; set; } = MixedRealityInputAction.None;
+
+        /// <summary>
+        /// Navigation action to use when events are raised.
+        /// </summary>
+        public static MixedRealityInputAction NavigationAction { get; set; } = MixedRealityInputAction.None;
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="useRailsNavigation">Should the gesture input source use rails navigation?</param>
+        public GestureInputSource(bool useRailsNavigation) : base("Gesture Input Source")
+        {
+            gestureRecognizer = new GestureRecognizer();
+
+            gestureRecognizer.Tapped += GestureRecognizer_Tapped;
+
+            gestureRecognizer.HoldStarted += GestureRecognizer_HoldStarted;
+            gestureRecognizer.HoldCompleted += GestureRecognizer_HoldCompleted;
+            gestureRecognizer.HoldCanceled += GestureRecognizer_HoldCanceled;
+
+            gestureRecognizer.ManipulationStarted += GestureRecognizer_ManipulationStarted;
+            gestureRecognizer.ManipulationUpdated += GestureRecognizer_ManipulationUpdated;
+            gestureRecognizer.ManipulationCompleted += GestureRecognizer_ManipulationCompleted;
+            gestureRecognizer.ManipulationCanceled += GestureRecognizer_ManipulationCanceled;
+
+            gestureRecognizer.SetRecognizableGestures(GestureSettings.Tap | GestureSettings.ManipulationTranslate | GestureSettings.Hold);
+
+            // We need a separate gesture recognizer for navigation, since it isn't compatible with manipulation
+            navigationGestureRecognizer = new GestureRecognizer();
+
+            navigationGestureRecognizer.NavigationStarted += NavigationGestureRecognizer_NavigationStarted;
+            navigationGestureRecognizer.NavigationUpdated += NavigationGestureRecognizer_NavigationUpdated;
+            navigationGestureRecognizer.NavigationCompleted += NavigationGestureRecognizer_NavigationCompleted;
+            navigationGestureRecognizer.NavigationCanceled += NavigationGestureRecognizer_NavigationCanceled;
+
+            if (useRailsNavigation)
+            {
+                navigationGestureRecognizer.SetRecognizableGestures(GestureSettings.NavigationRailsX | GestureSettings.NavigationRailsY | GestureSettings.NavigationRailsZ);
+            }
+            else
+            {
+                navigationGestureRecognizer.SetRecognizableGestures(GestureSettings.NavigationX | GestureSettings.NavigationY | GestureSettings.NavigationZ);
+            }
+
+            if (gestureRecognizer != null && !gestureRecognizer.IsCapturingGestures())
```

Comment:
allow the developer to defer the start of the recognizer

Reply:
no, either the system is enabled, or it's not. 

Same goes for the conversation we were having in the spatial awareness construct.

Reply:
i disagree here. there are times where the developer needs multiple recognizers to exist, but to be selective about which is running when (see Holograms 211 for a great example)

Reply:
Haha of course you do :p

Reply:
I believe we are having some (likely because of only talking via typing) confusion about what it means for a system vs a feature of that system to be enabled / activated. we should chat via phone or in shiproom (sometimes easier that way)

Reply:
Is this similar to how the dictation and keyword recognizers cannot run at the same time?